src/gluonts/shell/testutil.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+import json
+import socket
+import tempfile
+import time
+from contextlib import closing, contextmanager
+from multiprocessing import Process
+from pathlib import Path
+from typing import Any, ContextManager, Dict, Optional, Type
+
+# First-party imports
+from gluonts.dataset.repository.datasets import materialize_dataset
+from gluonts.model.predictor import Predictor
+from gluonts.shell.sagemaker import ServeEnv, ServePaths, TrainEnv, TrainPaths
+from gluonts.shell.sagemaker.params import encode_sagemaker_parameters
+from gluonts.shell.serve import ServerFacade, Settings, make_gunicorn_app
+
+
+def random_port() -> int:
+    """Returns a random unbound port."""
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
```

Comment:
I don't like single letter variables like `s`. Let's got with `socket_` or something like that here.